runtime/sema/checker.go

```diff
@@ -608,16 +601,9 @@ func (checker *Checker) checkTypeCompatibility(expression ast.Expression, valueT
 	case *ast.FixedPointExpression:
 		unwrappedTargetType := UnwrapOptionalType(targetType)
 
-		// If the target type is `Never`, the checks below will be performed
-		// (as `Never` is the subtype of all types), but the checks are not valid
-
-		if IsSubType(unwrappedTargetType, NeverType) {
-			break
-		}
-
 		valueTypeOK := CheckFixedPointLiteral(typedExpression, valueType, checker.report)
```

Comment:
The check above which was removed made this line conditional. Should this line now maybe be moved below new line 606, into the body of the if-statement?

Reply:
It wouldn't change the checker results, but yeah, can avoid additional computation if the condition fails. 
Refactored: e6be2687902bd3ea6cedefa01e6a36b260f4e6ae